core/src/main/java/org/elasticsearch/action/admin/indices/analyze/TransportAnalyzeAction.java

```diff
@@ -470,27 +422,156 @@ private void analyze(TokenStream stream, Analyzer analyzer, String field, Set<St
+                            throw new IllegalArgumentException("failed to find token filter under [" + tokenFilterName + "]");
+                        }
+                    }
+                }
+                if (tokenFilterFactories[i] == null) {
+                    throw new IllegalArgumentException("failed to find or create token filter under [" + tokenFilterName + "]");
+                }
+            }
+        }
+        return tokenFilterFactories;
+    }
+
+    private static TokenizerFactory parseTokenizerFactory(AnalyzeRequest request, AnalysisService analysisService, AnalysisRegistry analysisRegistry, Environment environment) throws IOException{
+        TokenizerFactory tokenizerFactory;
+        final String tokenizerName = request.tokenizer();
+        // parse custom settings if a value start "{"
+        if (tokenizerName.startsWith("{")) {
+            Settings settings = getCustomSettings(tokenizerName);
+            String tokenizerTypeName = settings.get("type");
+            if (tokenizerTypeName == null) {
+                throw new IllegalArgumentException("tokenizer type does not defined. [" + tokenizerName + "]");
+            }
+            AnalysisModule.AnalysisProvider<TokenizerFactory> tokenizerFactoryFactory =
+                analysisRegistry.getTokenizerProvider(tokenizerTypeName);
+            if (tokenizerFactoryFactory == null) {
+                throw new IllegalArgumentException("failed to find global tokenizer under [" + tokenizerTypeName + "]");
+            }
+            // Need to set custom "name" of tokenizer
+            tokenizerFactory = tokenizerFactoryFactory.get(getNaIndexSettings(settings), environment, "_custom_tokenizer", settings);
+        } else {
+            if (analysisService == null) {
+                AnalysisModule.AnalysisProvider<TokenizerFactory> tokenizerFactoryFactory = analysisRegistry.getTokenizerProvider(tokenizerName);
+                if (tokenizerFactoryFactory == null) {
+                    throw new IllegalArgumentException("failed to find global tokenizer under [" + tokenizerName + "]");
+                }
+                tokenizerFactory = tokenizerFactoryFactory.get(environment, tokenizerName);
+            } else {
+                tokenizerFactory = analysisService.tokenizer(tokenizerName);
+                if (tokenizerFactory == null) {
+                    throw new IllegalArgumentException("failed to find tokenizer under [" + tokenizerName + "]");
+                }
+            }
+        }
+        return tokenizerFactory;
+    }
+
+    private static IndexSettings getNaIndexSettings(Settings settings) {
+        IndexMetaData metaData = IndexMetaData.builder("_na_").settings(settings).build();
+        return new IndexSettings(metaData, Settings.EMPTY, Collections.emptyList());
+    }
+
+    private static Settings getCustomSettings(String providerSetting) {
+        return Settings.builder().loadFromSource(providerSetting)
+            // for _na_
```

Comment:
"_na_" is temporary index name. 
Because, IndexMetaData require the index name and these settings for build() method.
Of course, an analyzer should not care these settings.